src/porepy/models/solution_strategy.py

```diff
@@ -187,6 +184,9 @@ def __init__(self, params: Optional[dict] = None):
 
         """
 
+        self.nonlinear_solver_statistics: pp.SolverStatistics
```

Comment:
I agree that it would be sound to make it part of the NewtonSolver. Yet, as `self.nonlinear_solver_statistics` is meant to be the interface to any stats as number of nonlinear iterations, and e.g. the `DataSavingModelMixin` is using it, it has been made part of the solution strategy.